internal/engine/pod.go

```diff
@@ -102,6 +102,11 @@ func ensureManifestTargetWithPod(state *store.EngineState, pod *v1.Pod) (*store.
 	ns := k8s.NamespaceFromPod(pod)
 	hasSynclet := sidecar.PodSpecContainsSynclet(pod.Spec)
 
+	if pod.DeletionTimestamp != nil && !pod.DeletionTimestamp.IsZero() && ms.PodSet.Pods != nil {
```

Comment:
i thought we had a separate `prunePods` function for removing pods? does this logic make more sense there?